### PR TITLE
task 2.6

### PR DESCRIPTION
**Pull request by NguyenChanhTiep**

### CODE REVIEW

File: app/src/main/java/com/framgia/movie/ui/fragment/GenresFragment.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * A simple {@link Fragment} subclass.
+ */
+public class GenresFragment extends Fragment {
+
+    public GenresFragment() {
+        // Required empty public constructor
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/ui/fragment/GenresFragment.java]: ERROR: Line is longer than 100 characters (found 103).

File: app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+public class MainFragment extends Fragment {
+    private int mPosition;
+
+    public MainFragment() {
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java]: ERROR: Line is longer than 100 characters (found 103).

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -17,5 +17,13 @@
...
     <!-- view search-->
     <string name="title_search">Result</string>
+    <!-- title tab view pager-->
+    <string name="title_tab_popular">Popular</string>
+    <string name="title_tab_now_playing">Now Playing</string>
+    <string name="title_tab_upcoming">Upcoming</string>
+    <string name="title_tab_top_rate">Top Rate</string>
+    <string name="title_tab_genres">Genres</string>
+    <!-- TODO: Remove or change this placeholder text -->
+    <string name="hello_blank_fragment">Hello blank fragment</string>
```
Comment (le-quang-dao) [inline app/src/main/res/values/strings.xml]: WARNING: [UnusedResources] The resource `R.string.hello_blank_fragment` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/java/com/framgia/movie/ui/fragment/GenresFragment.java (hunk 1 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/fragment/GenresFragment.java]: fix

File: app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java (hunk 2 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java]: fix

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (NguyenChanhTiep) [reply app/src/main/res/values/strings.xml]: fix

File: app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import com.framgia.movie.ui.fragment.MainFragment;
+import com.framgia.movie.util.Constants;
+
+/**
+ * Created by TiepNguyen on 24-01-18.
+ * TiepNguyen8197@gmail.com.
+ */
+
+public class ViewPagerAdapter extends FragmentPagerAdapter {
+    private static final int TAB_NUMBER = 5;
```
Comment (ThuanPx) [inline app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java]: không cần thiết static đâu e

Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java]: fix

File: app/src/main/java/com/framgia/movie/ui/activity/MainActivity.java [hunk 5]
```diff
@@ -10,15 +11,18 @@
 import android.view.Menu;
 import android.view.MenuItem;
 import com.framgia.movie.R;
-import com.framgia.movie.ui.util.Constants;
+import com.framgia.movie.ui.adapter.ViewPagerAdapter;
+import com.framgia.movie.util.Constants;
 
-public class MainActivity extends AppCompatActivity implements SearchView.OnQueryTextListener{
+public class MainActivity extends AppCompatActivity implements SearchView.OnQueryTextListener {
+    private ViewPager mViewPager;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/movie/ui/activity/MainActivity.java]: nếu viewpager chỉ dùng trong menthod setViewPager thì e nên khai báo viewpager là local variable, k nên đặt là instance variable

File: app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+import com.framgia.movie.util.Constants;
+
+/**
+ * Created by TiepNguyen on 24-01-18.
+ * TiepNguyen8197@gmail.com.
+ */
+
+public class ViewPagerAdapter extends FragmentPagerAdapter {
+
+    private final int mTabNumber = 5;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java]: biến này là hằng số thì e nên khai báo static final

File: app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        super(fm);
+        mContext = context;
+    }
+
+    @Override
+    public Fragment getItem(int position) {
+        MainFragment mainFragment = new MainFragment();
+        Bundle bundle = new Bundle();
+        switch (position) {
+            case 0:
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java]: em nên sử dụng IntDef 
https://developer.android.com/reference/android/support/annotation/IntDef.html
để define type cho fragment như vậy sẽ clear hơn

File: app/src/main/java/com/framgia/movie/ui/activity/MainActivity.java (hunk 5 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/activity/MainActivity.java]: fix

File: app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java (hunk 6 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java]: fix

File: app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java (hunk 7 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java]: fix

File: app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        mPosition = getArguments().getInt(Constants.KEY_TAB_FRAGMENT_ARGUMENTS);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java]: ERROR: Line is longer than 100 characters (found 103).

File: app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+        super(fm);
+        mContext = context;
+    }
+
+    @Override
+    public Fragment getItem(int position) {
+        MainFragment mainFragment = new MainFragment();
+        Bundle bundle = new Bundle();
+        switch (position) {
+            case ItemTab.TAB_POPULAR:
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'ItemTab.TAB_POPULAR' due to existing static import 'com.framgia.movie.ui.adapter.ViewPagerAdapter.ItemTab.TAB_POPULAR'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+                bundle.putInt(Constants.KEY_TAB_FRAGMENT_ARGUMENTS, ItemTab.TAB_POPULAR);
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'ItemTab.TAB_POPULAR' due to existing static import 'com.framgia.movie.ui.adapter.ViewPagerAdapter.ItemTab.TAB_POPULAR'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+                mainFragment.setArguments(bundle);
+                return mainFragment;
+            case ItemTab.TAB_NOW_PLAYING:
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'ItemTab.TAB_NOW_PLAYING' due to existing static import 'com.framgia.movie.ui.adapter.ViewPagerAdapter.ItemTab.TAB_NOW_PLAYING'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+                bundle.putInt(Constants.KEY_TAB_FRAGMENT_ARGUMENTS, ItemTab.TAB_NOW_PLAYING);
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'ItemTab.TAB_NOW_PLAYING' due to existing static import 'com.framgia.movie.ui.adapter.ViewPagerAdapter.ItemTab.TAB_NOW_PLAYING'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+                mainFragment.setArguments(bundle);
+                return mainFragment;
+            case ItemTab.TAB_UPCOMING:
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'ItemTab.TAB_UPCOMING' due to existing static import 'com.framgia.movie.ui.adapter.ViewPagerAdapter.ItemTab.TAB_UPCOMING'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+                bundle.putInt(Constants.KEY_TAB_FRAGMENT_ARGUMENTS, ItemTab.TAB_UPCOMING);
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'ItemTab.TAB_UPCOMING' due to existing static import 'com.framgia.movie.ui.adapter.ViewPagerAdapter.ItemTab.TAB_UPCOMING'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+                mainFragment.setArguments(bundle);
+                return mainFragment;
+            case ItemTab.TAB_TOP_RATE:
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'ItemTab.TAB_TOP_RATE' due to existing static import 'com.framgia.movie.ui.adapter.ViewPagerAdapter.ItemTab.TAB_TOP_RATE'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+                bundle.putInt(Constants.KEY_TAB_FRAGMENT_ARGUMENTS, ItemTab.TAB_TOP_RATE);
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'ItemTab.TAB_TOP_RATE' due to existing static import 'com.framgia.movie.ui.adapter.ViewPagerAdapter.ItemTab.TAB_TOP_RATE'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java [hunk 17]
```diff
@@ -0,0 +1,94 @@
...
+                mainFragment.setArguments(bundle);
+                return mainFragment;
+            case ItemTab.TAB_GENRES:
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java]: WARNING: [UnnecessaryFullyQualifiedName] Unnecessary use of fully qualified name 'ItemTab.TAB_GENRES' due to existing static import 'com.framgia.movie.ui.adapter.ViewPagerAdapter.ItemTab.TAB_GENRES'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnnecessaryFullyQualifiedName

File: app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java (hunk 8 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java]: fix

File: app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java (hunk 9 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java]: fix

File: app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java (hunk 10 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java]: fix

File: app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java (hunk 11 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java]: fix

File: app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java (hunk 12 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java]: fix

File: app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java (hunk 13 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java]: fix

File: app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java (hunk 14 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java]: fix

File: app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java (hunk 15 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java]: fix

File: app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java (hunk 16 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java]: fix

File: app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java (hunk 17 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java]: fix

File: app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java [hunk 18]
```diff
@@ -0,0 +1,94 @@
...
+
+    @Override
+    public int getCount() {
+        return TAB_NUMBER;
+    }
+
+    @Override
+    public CharSequence getPageTitle(int position) {
+        switch (position) {
+            case 0:
```
Comment (ThuanPx) [inline app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java]: làm như ở getItem luôn e

Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/adapter/ViewPagerAdapter.java]: fix

File: app/build.gradle [hunk 19]
```diff
@@ -28,6 +28,7 @@ android {
 }
 
 dependencies {
+  implementation 'com.android.support:support-v4:26.1.0'
   compile fileTree(include: ['*.jar'], dir: 'libs')
```
Comment (daolq3012) [inline app/build.gradle]: pls change ``compile`` to ``implementation`` fo all variable
declare in block ``dependencies``

File: app/src/main/java/com/framgia/movie/ui/fragment/GenresFragment.java [hunk 20]
```diff
@@ -0,0 +1,25 @@
...
+import android.view.View;
+import android.view.ViewGroup;
+import com.framgia.movie.R;
+
+/**
+ * A simple {@link Fragment} subclass.
+ */
+public class GenresFragment extends Fragment {
+
+    public GenresFragment() {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/movie/ui/fragment/GenresFragment.java]: remove this constructor method
It unnecessary!

File: app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java [hunk 21]
```diff
@@ -0,0 +1,34 @@
...
+import com.framgia.movie.util.Constants;
+
+/**
+ * A simple {@link Fragment} subclass.
+ */
+
+public class MainFragment extends Fragment {
+    private int mPosition;
+
+    public MainFragment() {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java]: like above

File: app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java [hunk 22]
```diff
@@ -0,0 +1,34 @@
...
+import android.view.ViewGroup;
+import com.framgia.movie.R;
+import com.framgia.movie.util.Constants;
+
+/**
+ * A simple {@link Fragment} subclass.
+ */
+
+public class MainFragment extends Fragment {
+    private int mPosition;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java]: why we need this variable?

File: app/src/main/java/com/framgia/movie/util/Constants.java [hunk 23]
```diff
@@ -1,14 +1,15 @@
...
  */
 
 public final class Constants {
-    
+
     private Constants() {
     }
 
     public static final String KEY_SEARCH = "key_search";
+    public static final String KEY_TAB_FRAGMENT_ARGUMENTS = "key_arguments";
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/movie/util/Constants.java]: change to
```java
public static final String KEY_TAB_FRAGMENT_ARGUMENTS = "KEY_TAB_FRAGMENT_ARGUMENTS";
```

File: app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java [hunk 24]
```diff
@@ -0,0 +1,29 @@
+package com.framgia.movie.ui.fragment;
+
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import com.framgia.movie.R;
+import com.framgia.movie.util.Constants;
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java]: ERROR: Unused import: com.framgia.movie.util.Constants. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

Comment (le-quang-dao) [inline app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.framgia.movie.util.Constants'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/build.gradle (hunk 19 above)
Comment (NguyenChanhTiep) [reply app/build.gradle]: fix

File: app/src/main/java/com/framgia/movie/ui/fragment/GenresFragment.java (hunk 20 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/fragment/GenresFragment.java]: fix

File: app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java (hunk 22 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java]: fix

File: app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java (hunk 21 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java]: fix

File: app/src/main/java/com/framgia/movie/util/Constants.java (hunk 23 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/util/Constants.java]: fix

File: app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java (hunk 24 above)
Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java]: fix

Comment (NguyenChanhTiep) [reply app/src/main/java/com/framgia/movie/ui/fragment/MainFragment.java]: fix

**Pull request merged**